doc/build.md

```diff
@@ -2,121 +2,193 @@
+#### 1. Tools and System Prerequisites
 
-#### 1. Install prerequisites
+Before building the project, consult the following table to ensure you use the recommended version of compiler, operating system, cmake, etc. 
 
-- Required:
-  - C++ compiler:
-       - GCC
-       - Xcode
-       - Microsoft Visual Studio
-  - CMake 
-  - Python
-- Optional
-  - Ninja
+|        Required       | ![](images/windows.png)   |                            ![](images/mac.png)               |   ![](images/linux.png)     |
+|:---------------------:|:-------------------------:|:------------------------------------------------------------:|:---------------------------:|
+|    Operating System   |         Windows 10        | Catalina (10.15), Mojave (10.14), High Sierra (10.13)        |       CentOS 7              |
+| Minimum Compiler Version  | Microsoft VS 2017 or 2015 |                      Clang 9.0                           |       GCC 6.3.1             |
+| Minimum Cmake Version |           3.13            |                             3.13                             |         3.13                |
+|         Python        |           2.7.15          |                            2.7.15                            |        2.7.15               |
+|    Python Packages    | PyYAML, PySide, PyOpenGL, Jinja2        | PyYAML, PySide2, PyOpenGL, Jinja2              |      PyYAML, PySide, PyOpenGL, Jinja2             |
+|    Build generator    | Visual Studio, Ninja (Recommended)    |  XCode, Ninja(Recommended)                       |    Ninja(Recommended)       |
+|    Command processor  | Visual Studio X64 2015 or 2017 command prompt  |                     bash                |             bash            |
+| Supported Maya Version|      2018, 2019, 2020           |                      2018, 2019, 2020                  |        2018,2019, 2020      | 
 
-#### 2. Download the source code
+***NOTE:*** We haven't fully tested the plug-ins on ```Catalina``` and it is still at the experimental stage.
+
+#### 2. Download and Build Pixar USD 
+
+See Pixar's official github page for instructions on how to build USD: https://github.com/PixarAnimationStudios/USD
+
+For additional information on building USD, see the ***Building Pixar USD*** section below.
+
+***NOTE:*** Make sure that you don't have older USD locations in your ```PATH``` and ```PYTHONPATH``` environment settings. ```PATH``` and ```PYTHONPATH``` are automatically adjusted inside the project to point to the correct USD location. See ```cmake/usd.cmake```.
+
+#### 3. Download Universal Front End (UFE) 
+
+The Universal Front End (UFE) is a DCC-agnostic component that allows Maya to browse and edit data in multiple data models. This allows Maya to edit pipeline data such as USD.
+UFE is developed as a separate binary component, and is therefore versioned separately from Maya. UFE is installed as a built-in component of Maya.
```

Comment:
This reads a bit odd. It reads as if you need to download the full UFE technology for Maya but that hasn't been the case before, has that changed? If so, where's the download for that binary component?

How about changing:

> UFE is developed as a separate binary component, and is therefore versioned separately from Maya. UFE is installed as a built-in component of Maya.

To:

 > UFE comes installed as a built-in component with Maya. However UFE is developed as a separate binary component, and is therefore versioned separately from Maya. To enable the Maya UFE support you will need to link to the UFE headers included in the Maya Devkit.

Of course correct me if I'm wrong. ;)

Also, if UFE has its own versioning is it supposed to be in the compatibility list at the top too? Maybe not if they are all compatible.